regions/io/core.py

```diff
@@ -221,3 +371,120 @@ def to_region(self):
 
     def _raise_error(self, msg):
         raise self.error[self.format_type](msg)
+
+    def to_crtf(self):
+        """
+        Makes shape CRTF compatible.
+        """
+        self.format_type = 'CRTF'
+
+    def to_ds9(self):
+        """
+        Makes shape DS9 compatible
+        """
+        self.format_type = 'DS9'
+
+    def _validate(self):
+        """
+        Checks whether all the attributes of this object is valid according to the given format.
+        """
+        if self.format_type not in ['CRTF', 'DS9']:
+            raise ValueError("{0} is not available as io".format(self.format_type))
+        if self.region_type not in regions_attributes.keys():
+            raise ValueError("{0} is not a valid region type in this package".format(self.region_type, self.format_type))
+        if self.coordsys not in valid_coordsys[self.format_type]:
+            raise ValueError("{0} is not a valid coordinate reference frame in astropy".
+                             format(self.coordsys, self.format_type))
+
+
+def to_shape_list(region_list, format_type='DS9', coordinate_system='fk5'):
+
+    shape_list = ShapeList()
+
+    for region in region_list:
+
+        coord = []
+        reg_type = str(type(region)).split(".")[2]
+
+        for val in regions_attributes[reg_type]:
+            coord.append(getattr(region, val))
+
+        if reg_type == 'polygon':
+            coord = [x for x in region.vertices]
+
+        coordsys = 'image'
+        if isinstance(region, SkyRegion):
+            coordsys = coord[0].name
+
+        coordsys = coordinate_system or coordsys
+        frame = coordinates.frame_transform_graph.lookup_name(coordsys)
+
+        new_coord = []
+        for val in coord:
+            if isinstance(val, Angle) or isinstance(val, u.Quantity) or isinstance(val, float):
+                new_coord.append(val)
+            elif isinstance(val, PixCoord):
+                new_coord.append(u.Quantity(val.x, u.dimensionless_unscaled))
+                new_coord.append(u.Quantity(val.y, u.dimensionless_unscaled))
+            else:
+                new_coord.append(Angle(val.transform_to(frame).spherical.lon))
+                new_coord.append(Angle(val.transform_to(frame).spherical.lat))
+
+        if format_type == 'DS9':
+            meta = to_ds9_meta(region.meta, region.visual)
+        else:
```

Comment:
probably want `elif format_type=='CRTF'` here, and have the final `else` raise an exception

Reply:
format_type can be either 'CRTF' or 'DS9', it is restricted in it's setter method.